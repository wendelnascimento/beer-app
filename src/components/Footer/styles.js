import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 16px;
  color: #cccccc;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
`;
