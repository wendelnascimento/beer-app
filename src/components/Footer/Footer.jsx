import React from 'react';

import { Wrapper } from './styles';

const Footer = () => (
  <Wrapper>
    <span>
      Beer App
      {' '}
      <span role="img" aria-label="Beer">
        🍺
      </span>
      {' '}
      - ©
      {' '}
      {new Date().getFullYear()}
    </span>
  </Wrapper>
);

export default Footer;
