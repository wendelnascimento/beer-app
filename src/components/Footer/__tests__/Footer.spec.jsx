import React from 'react';
import { shallow, mount } from 'enzyme';

import Footer from '../Footer';

describe('<Footer />', () => {
  it('does render properly', () => {
    shallow(<Footer />);
  });

  it('matches stored snapshot', () => {
    const wrapper = mount(<Footer />);

    expect(wrapper).toMatchSnapshot();
  });

  it('does have current year after copyright', () => {
    const wrapper = mount(<Footer />);

    expect(
      wrapper
        .find('span')
        .first()
        .text(),
    ).toBe(`Beer App 🍺 - © ${new Date().getFullYear()}`);
  });
});
