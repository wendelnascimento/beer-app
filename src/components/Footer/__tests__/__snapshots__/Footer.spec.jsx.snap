// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches stored snapshot 1`] = `
.c0 {
  padding: 16px;
  color: #cccccc;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
}

<Footer>
  <styled.footer>
    <footer
      className="c0"
    >
      <span>
        Beer App
         
        <span
          aria-label="Beer"
          role="img"
        >
          üç∫
        </span>
         
        - ¬©
         
        2020
      </span>
    </footer>
  </styled.footer>
</Footer>
`;
